index.js

```diff
@@ -80,25 +80,25 @@ module.exports = {
 
     // Let's try to match https based repo url
     let match = url.match(
-      /https?:\/\/(www\.)?(github|gitlab).com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
+      /https?:\/\/(www\.)?(?<server>[\w.-]+).com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
     );
     if (match && match.groups && match.groups.owner && match.groups.name) 
-      return !returnUrl ? `${match.groups.name}` : `${match[0]}`;
+      return returnUrl ? `https://${match.groups.server}.com/${match.groups.owner}/${match.groups.name}` : `${match.groups.name}`;
 
     // Now trying to match ssh based repo url
     match = url.match(
       /git@(?<server>[\w.-]+).com:(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
     );
     if (match && match.groups && match.groups.owner && match.groups.name) 
```

Comment:
Maybe extract that boolean to a variable like "const isMatchComplete = match && match.groups && match.groups.owner && match.groups.name && match.groups.owner" would be great

Reply:
Sounds good